packages/key-management/src/types.ts

```diff
@@ -222,12 +222,13 @@ export interface Witnesser {
   /**
    * Generates the witness data for a given transaction.
    *
-   * @param txInternals The transaction body along with its hash for which the witness data is to be generated.
+   * @param transaction The transaction along with its hash for which the witness data is to be generated.
+   * @param context The witness sign transaction context
    * @param options Optional additional parameters that may influence how the witness data is generated.
    * @returns A promise that resolves to the generated witness data for the transaction.
    */
   witness(
-    txInternals: Cardano.TxBodyWithHash,
+    transaction: Serialization.Transaction,
```

Comment:
Would've been nice to do this refactor in a separate commit, please see if you can do this without too much effort. Otherwise feel free to resolve this change request